File: web/projects/portal/src/app/composer/gui/vs/objects/event/change-vs-object-binding-event.ts
/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { VSObjectEvent } from "../../../../../vsobjects/event/vs-object-event";
import { AssetEntry } from "../../../../../../../../shared/data/asset-entry";
import { TableTransfer } from "../../../../../common/data/dnd-transfer";

/**
 * Event used to change the binding of an object.
 */
export class ChangeVSObjectBindingEvent extends VSObjectEvent {
   /**
    * The AssetEntry of the datasource to bind to;
    */
   public binding: AssetEntry[];

   /**
    * The information for binding dropped from a component (tables only)
    */
   public componentBinding: TableTransfer;

   /**
    * The new x position of item if creating new one;
    */
   public x: number;

   /**
    * The new y position of item if creating new one;
    */
   public y: number;

   /**
    * Whether or not to place new object into tab.
    */
   public tab: boolean;

   /**
    * Creates a new instance of <tt>ChangeVSObjectBindingEvent</tt>.
    *
    * @param objectName the name of the object.
    */
   constructor(objectName: string)
   {
      super(objectName);
   }
}

File: core/src/main/java/inetsoft/web/composer/vs/objects/event/ChangeVSObjectBindingEvent.java
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package inetsoft.web.composer.vs.objects.event;

import inetsoft.uql.asset.AssetEntry;
import inetsoft.web.binding.dnd.TableTransfer;

import java.util.List;

/**
 * Class that encapsulates the parameters for changing the binding of an object.
 *
 * @since 12.3
 */
public class ChangeVSObjectBindingEvent extends VSObjectEvent {
   public List<AssetEntry> getBinding() {
      return binding;
   }

   public void setBinding(List<AssetEntry> binding) {
      this.binding = binding;
   }

   public TableTransfer getComponentBinding() {
      return componentBinding;
   }

   public void setComponentBinding(TableTransfer componentBinding) {
      this.componentBinding = componentBinding;
   }

   public int getX() {
      return x;
   }

   public void setX(int x) {
      this.x = x;
   }

   public int getY() {
      return y;
   }

   public void setY(int y) {
      this.y = y;
   }

   public boolean isTab() {
      return tab;
   }

   public void setTab(boolean tab) {
      this.tab = tab;
   }

   @Override
   public String toString() {
      return "ChangeVSObjectBindingEvent{" +
         "name='" + this.getName() + '\'' +
         "binding='" + binding + '\'' +
         "componentBinding='" + componentBinding + '\'' +
         "x='" + x + '\'' +
         "y='" + y + '\'' +
         "tab='" + tab + '\'' +
         '}';
   }

   private List<AssetEntry> binding;
   private TableTransfer componentBinding;
   private int x;
   private int y;
   private boolean tab;
}
